pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -216,17 +216,26 @@ func podToEndpointAddressForService(svc *v1.Service, pod *v1.Pod) (*v1.EndpointA
 		return podToEndpointAddress(pod), nil
 	}
 
-	// api-server service controller ensured that the service got the correct IP Family
-	// according to user setup, here we only need to match EndPoint IPs' family to service
-	// actual IP family. as in, we don't need to check service.IPFamily
+	// for non-headless services, the api-server service controller ensured that the service got the
+	// correct IP Family according to user setup, here we only need to match EndPoint IPs' family
+	// to service actual IP family.
+
+	isIPv6Service := utilnet.IsIPv6String(svc.Spec.ClusterIP)
+
+	// for headless services ClusterIP is not set, so we filter by the spec.IPFamily field
+	// if spec.IPFamily is nil it defaults to IPv4
+	// TODO for headless services without spec.IPFamily it should be possible to return
```

Comment:
What do we think would prevent this from being feasible?  I don't think we have ever (intentionally) mixed IP families in a single Endpoints, which will almost certainly explode someone, I suppose.